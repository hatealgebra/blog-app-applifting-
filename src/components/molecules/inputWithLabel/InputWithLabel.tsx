import React, { Dispatch, DispatchWithoutAction } from 'react';
import { StyledTextInput, TSize } from '../../atoms/input/input.styled';

import { StyledInputWithLabel } from './inputWithLabel.styled';

const InputWithLabel = ({
  label,
  value,
  onChange,
  type,
  placeholder,
  ...props
}: InputWithLabelProps) => {
  return (
    <StyledInputWithLabel>
      <label htmlFor={label}>{label}</label>
      <StyledTextInput
        value={value}
        onChange={onChange}
        placeholder={placeholder}
        type={type}
        id={label}
      />
    </StyledInputWithLabel>
  );
};

type TTypeInput = 'text' | 'password' | 'email';

export interface InputWithLabelProps {
  type?: TTypeInput;
  size?: TSize;
  placeholder?: string;
  value: any;
  onChange: Dispatch<any>;
  label: string;
}

export default InputWithLabel;
