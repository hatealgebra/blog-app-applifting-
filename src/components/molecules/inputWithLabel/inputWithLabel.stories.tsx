import React from 'react';

import InputWithLabel from './InputWithLabel';

export const InputWithLabelTypes = () => {
  const [name, setName] = React.useState('hello');
  const [email, setEmail] = React.useState('');
  const [pwd, setPwd] = React.useState('');

  return (
    <div style={{ display: 'flex', flexDirection: 'column', gap: '20px' }}>
      <InputWithLabel
        label="Username"
        placeholder="Enter your name"
        value={name}
        onChange={(e) => setName(e.target.value)}
      />
      <InputWithLabel
        label="Email"
        placeholder="me@example.com"
        value={email}
        onChange={(e) => setEmail(e.target.value)}
      />
      <InputWithLabel
        label="Password"
        placeholder="Enter your password"
        value={pwd}
        onChange={(e) => setPwd(e.target.value)}
      />
    </div>
  );
};

export default {
  title: 'Molecules/Input with label',
  component: InputWithLabel,
};
