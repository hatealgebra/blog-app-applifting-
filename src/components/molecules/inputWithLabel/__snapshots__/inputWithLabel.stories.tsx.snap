// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecules/Input with label InputWithLabelTypes smoke-test 1`] = `
<div style="display: flex; flex-direction: column; gap: 20px;">
  <div class="sc-jSwlEQ eeEroX">
    <label for="Username">
      Username
    </label>
    <input placeholder="Enter your name"
           id="Username"
           class="sc-hLclGa iSKFPv"
           value="hello"
    >
  </div>
  <div class="sc-jSwlEQ eeEroX">
    <label for="Email">
      Email
    </label>
    <input placeholder="me@example.com"
           id="Email"
           class="sc-hLclGa iSKFPv"
           value
    >
  </div>
  <div class="sc-jSwlEQ eeEroX">
    <label for="Password">
      Password
    </label>
    <input placeholder="Enter your password"
           id="Password"
           class="sc-hLclGa iSKFPv"
           value
    >
  </div>
</div>
`;
